models/Translation/Dao.php

```diff
@@ -40,12 +40,20 @@ public function getDatabaseTableName(): string
 
     /**
      * @param string $key
+     * @param ?array $languages
      *
      * @throws NotFoundResourceException
      */
-    public function getByKey($key)
+    public function getByKey($key, $languages = null)
     {
-        $data = $this->db->fetchAll('SELECT * FROM ' . $this->getDatabaseTableName() . ' WHERE `key` = ? ORDER BY `creationDate` ', [$key]);
+        if (is_array($languages)) {
+            $sql = 'SELECT * FROM ' . $this->getDatabaseTableName() . ' WHERE `key` = ?
+            AND `language` IN (' . implode(',' , array_map([$this->db, 'quote'], $languages)) . ') ORDER BY `creationDate` ';
```

Comment:
Can we please stop inlining parameters and use prepared statements instead?

Reply:
that was my first try but it is not working with IN operator and this is a general improvement topic as we do this at lot of places.

Reply:
Are you sure it doesn't work? I'm pretty sure I already used it. I think it has to be an array and needs a type, like: `fetchAll('...', ['key', ['de','en']], [1 => \Doctrine\DBAL\Connection::PARAM_STR_ARRAY])` (it would probably better to use named parameters instead of positional ones in this case). See: https://www.doctrine-project.org/projects/doctrine-dbal/en/2.13/reference/data-retrieval-and-manipulation.html#list-of-parameters-conversion

Reply:
yes, you are right. I missed passing the type in first attempt. I'll give it a try. thanks!

Reply:
@jdreesen works like a charm. thanks!